lib/puppet-syntax/manifests.rb

```diff
@@ -15,7 +15,7 @@ def check(filelist)
       end
 
       # Catch syntax warnings.
-      Puppet::Util::Log.newdestination(Puppet::Test::LogCollector.new(errors))
+      Puppet::Util::Log.newdestination(Puppet::Test::LogCollector.new(output))
       Puppet::Util::Log.level = :warning
```

Comment:
I think there might be a way to determine the type each of log entry which would allow us to better distinguish warnings, notices and errors.

This PR stores deprecation notices in a `warnings` array, though strictly I think these are notices rather than warnings.

I'll work on it and see what I can come up with.